Drivers/Periphs/Src/IMU.c

```diff
@@ -36,5 +57,76 @@ BaseType_t IMU_Init(void) {
  * @return pdTRUE if data was successfully fetched from queue, pdFALSE if queue is empty
  */
 BaseType_t IMU_ReadData(IMUData_t *Data) {
-  // TODO: Read data from queue
+  int pTxData = 0;
```

Comment:
this seems quite a bit more complex than it needs to be (along with the other IMU read function); i'd recommend rechecking the implementation I sent and taking another look at how to retrieve data from the IMU. 

however, i'm hella biased towards liking my own implementation so if this works then no complaints. just know that the way this is written makes it very difficult to debug (and read)